### PR TITLE
Reduce the default maximum heap size for all Scala processes.

### PR DESCRIPTION
**Pull request by SamirTalwar-DA**
The JVM, as of version 8, on a 64-bit system, defaults to a quarter of your total memory as the maximum heap size. Most developers hover between 8 and 32 GB of RAM in their computer, and our CI has 32 GB available. This means that we might be allocating 8 GB of heap per Java process.

This PR decreases that to a maximum of 2 GB (overridable). For Sandbox, this is plenty, but it should mean more garbage collections are triggered, keeping it lean and allowing it to run in parallel with everything else going on.

I'm also decreasing `scalac`'s heap size, for the same reason. I've increased its stack size to 2 MB because the codegen sample app apparently needs it, and there's no harm in increasing it for everything AFAIK.

I also deleted `da_scala_macro_library`, because it's unused and I didn't want to have to figure out how this stuff applies there.

Hopefully this makes our Windows builds happier.

### Pull Request Checklist

- [x] Read and understand the [contribution guidelines](https://github.com/digital-asset/daml/blob/master/CONTRIBUTING.md)
- [x] Include appropriate tests
- [x] Set a descriptive title and thorough description
- [ ] Add a reference to the [issue this PR will solve](https://github.com/digital-asset/daml/issues), if appropriate
- [x] Include changelog additions in one or more commit message bodies between the `CHANGELOG_BEGIN` and `CHANGELOG_END` tags
- [ ] Normal production system change, include purpose of change in description

NOTE: CI is not automatically run on non-members pull-requests for security
reasons. The reviewer will have to comment with `/AzurePipelines run` to
trigger the build.

### CODE REVIEW

File: bazel_tools/scala.bzl [hunk 1]
```diff
@@ -442,30 +482,20 @@ def da_scala_library_suite(name, **kwargs):
...
-    And forwards to `scala_macro_library` from `rules_scala`.
-    Refer to the [`rules_scala` documentation][rules_scala_docs].
-
-    [rules_scala_docs]: https://github.com/bazelbuild/rules_scala#scala_library
-    """
-    _wrap_rule(scala_macro_library, **kwargs)
-    _create_scala_source_jar(**kwargs)
-
-def da_scala_binary(name, unused_dependency_checker_mode = "error", **kwargs):
+def da_scala_binary(name, initial_heap_size = default_initial_heap_size, max_heap_size = default_max_heap_size, **kwargs):
```
Comment (stefanobaghino-da) [inline bazel_tools/scala.bzl:498]: Probably keeping the docstring makes sense: it links the documentation for the underlying rule and explains why that would be important.

Comment (SamirTalwar-DA) [reply bazel_tools/scala.bzl:498]: I also deleted the function that went with this docstring. 🙂

File: BAZEL-JVM.md [hunk 2]
```diff
@@ -495,19 +495,8 @@ Strive to make library targets as small as possible. In a situation where
...
-compiler errors of the following form (formatted for readability):
-
-```
-error: macro annotation could not be expanded (the most common reason
-for that is that you need to enable the macro paradise plugin; another
-possibility is that you try to use macro annotation in the same
-compilation run that defines it)
-```
+attributes as `da_scala_library`, but enables `unused_dependency_checker_mode`
+by default.
```
Comment (aherrmann-da) [inline BAZEL-JVM.md:499]: I don't think that's true. [`unused_dependency_checker_mode` is set to `False` within `rules_scala` itself](https://github.com/bazelbuild/rules_scala/blob/6c16cff213b76a4126bdc850956046da5db1daaa/scala/private/rules/scala_library.bzl#L118).
```suggestion
attributes as `da_scala_library` with the exception of
`unused_dependency_checker_mode` which will always be disabled.
```

Comment (SamirTalwar-DA) [reply BAZEL-JVM.md:499]: Aha, sorry, I misread it and thought it was about `da_scala_library`. I'll revert the text.

Comment (SamirTalwar-DA) [reply BAZEL-JVM.md]: Somehow I missed your suggestion and just did it manually. Not sure why. Anyway, done. 🙂

File: navigator/backend/BUILD.bazel [hunk 3]
```diff
@@ -14,10 +14,6 @@ navigator_scalacopts = [
     "-Xsource:2.13",
 ]
 
-common_jvm_flags = [
```
Comment (cocreature) [inline navigator/backend/BUILD.bazel:17]: This was just unused right?

Comment (SamirTalwar-DA) [reply navigator/backend/BUILD.bazel:17]: Yup yup.

File: BAZEL-JVM.md (hunk 2 above)
Comment (aherrmann-da) [reply BAZEL-JVM.md]: Thank you!

**Pull request merged**